vdf/src/gmp_classgroup/mod.rs

```diff
@@ -200,13 +336,19 @@ impl Default for GmpClassGroup {
 }
 
 impl MulAssign<Self> for GmpClassGroup {
+    #[cfg_attr(not(debug_assertions), inline(always))]
```

Comment:
Maybe the number of `Mul[Assign]` impls could be halved by using `Borrow`—that should cover both the reference and value cases:

```rust
impl<B: Borrow<GmpClassGroup>> MulAssign<B> for GmpClassGroup { … }
```